src/main/java/frc/robot/commands/GoToPose.java

```diff
@@ -0,0 +1,104 @@
+package frc.robot.commands;
+
+import java.util.function.Supplier;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.TestConstants;
+import frc.robot.subsystems.Drivetrain;
+
+/**
+ * Tests the odometry of the robot by driving a certain distance and calculating the error.
+ */
+public class GoToPose extends CommandBase {
+
+  private Drivetrain m_drive; 
+  private Supplier<Pose2d> m_poseSupplier;
+  private double m_startTime;
+  private Pose2d m_finalPose;
+  private Pose2d m_error;
+  private boolean m_relativeToRobot;
+  private boolean m_doNotEnd = false;
+  
+  /**
+   * Returns a command that goes to a position based on shuffleboard inputs
+   * DO NOT USE IF SHUFFLEBOARD IS DISABLED 
+   * @param drive drivetrain to be used for the command
+   * @param relativeToRobot should pose be converted to a transformation
+   * @param doNotEnd should the robot not end for tunning purposes
+   */
+  public GoToPose(Drivetrain drive, boolean relativeToRobot, boolean doNotEnd){
```

Comment:
You can use PerpetualCommand(commandToRun) which will run the command but ignore the isFinished, instead of using the extra boolean param. Or, you can add the .perpetually() decorator to do the same thing

Reply:
Basically that would remove the doesNotEnd parameter

Reply:
Perpertual Command is deprecated and will be removed in the future so I don't think we should use it.